File: app/src/main/java/com/dumuzeyn/mp3player/TrackDeletionPolicy.java
package com.dumuzeyn.mp3player;

import android.net.Uri;

final class TrackDeletionPolicy {
    private TrackDeletionPolicy() {
    }

    static boolean isMediaStore(Uri uri) {
        return uri != null && isMediaStore(uri.toString());
    }

    static boolean isContentUri(Uri uri) {
        return uri != null && isContentUri(uri.toString());
    }

    static boolean isMediaStore(String uri) {
        return uri != null && uri.startsWith("content://media/");
    }

    static boolean isContentUri(String uri) {
        return uri != null && uri.startsWith("content://");
    }
}

File: app/src/main/java/com/dumuzeyn/mp3player/TrackDeletionPolicy.kt
package com.dumuzeyn.mp3player

import android.net.Uri

object TrackDeletionPolicy {
    @JvmStatic
    fun isMediaStore(uri: Uri?): Boolean = uri != null && isMediaStore(uri.toString())

    @JvmStatic
    fun isContentUri(uri: Uri?): Boolean = uri != null && isContentUri(uri.toString())

    @JvmStatic
    fun isMediaStore(uri: String?): Boolean = uri?.startsWith("content://media/") == true

    @JvmStatic
    fun isContentUri(uri: String?): Boolean = uri?.startsWith("content://") == true
}
